storage/grpc_client.go

```diff
@@ -156,8 +157,52 @@ func (c *grpcStorageClient) CreateBucket(ctx context.Context, project string, at
 	return battrs, err
 }
 
-func (c *grpcStorageClient) ListBuckets(ctx context.Context, project string, opts ...storageOption) (*BucketIterator, error) {
-	return nil, errMethodNotSupported
+func (c *grpcStorageClient) ListBuckets(ctx context.Context, project string, opts ...storageOption) *BucketIterator {
+	s := callSettings(c.settings, opts...)
+	it := &BucketIterator{
+		ctx:       ctx,
+		projectID: project,
+	}
+
+	var gitr *gapic.BucketIterator
+	fetch := func(pageSize int, pageToken string) (token string, err error) {
```

Comment:
I'm wondering if, to integrate this with the production code path, we should make `fetch` a field on the iterator rather than a method? Or can we just remove the fetch method entirely? It might be good to include a TODO about how you foresee this integrating because it's not totally obvious to me.

Reply:
> Or can we just remove the fetch method entirely? It might be good to include a TODO about how you foresee this integrating...

That was exactly my thinking and I will add a comment to that affect.

The existing `fetch` method references some fields of the `BucketIterator` i.e. the `Prefix`, but we can capture that reference in the function closure created here when we define `fetch`. Also, the existing  `BucketIterator.fetch` is just supplied to `iterator.NewPageInfo` and not referenced elsewhere. Making it a field is not entirely necessary.

Reply:
Added a comment to `fetch`. Done.